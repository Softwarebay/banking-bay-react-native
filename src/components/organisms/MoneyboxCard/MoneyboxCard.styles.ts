import { StyleSheet } from 'react-native';
import { colors } from 'theme';

export const styles = StyleSheet.create({
	container: {
		alignItems: 'stretch',
		padding: 20,
	},
	currency: {
		fontFamily: 'MulishBold',
		marginLeft: 5,
	},
	extendButton: {
		backgroundColor: colors.coral,
		flex: 1,
		marginRight: 5,
	},
	floatStyle: {
		fontFamily: 'MulishRegular',
		fontSize: 17,
		lineHeight: 22.1,
	},
	goal: {
		marginLeft: 'auto',
	},
	goalTitle: {
		fontSize: 15,
	},
	iconBackground: {
		alignItems: 'center',
		backgroundColor: colors.green,
		borderRadius: 50,
		height: 27,
		justifyContent: 'center',
		width: 27,
	},
	iconContainer: {
		alignSelf: 'flex-start',
		marginRight: 10,
	},
	intStyle: {
		alignSelf: 'flex-start',
		fontFamily: 'MulishRegular',
		fontSize: 22,
		lineHeight: 22.1,
	},
	progressBar: {
		marginTop: 10,
	},
	saved: {
		justifyContent: 'center',
		marginBottom: 10,
		marginTop: 5,
	},
	savedCurrency: {
		top: 1,
	},
	savedFloat: {
		color: colors.bodyText,
		fontSize: 12,
	},
	savedFloatNum: {
		marginRight: 5,
	},
	savedInt: {
		color: colors.bodyText,
		fontSize: 15,
	},
	topUpButton: {
		backgroundColor: colors.blue,
		flex: 1,
		marginRight: 5,
	},
	withdrawalButton: {
		flex: 1,
		marginLeft: 5,
	},
});
