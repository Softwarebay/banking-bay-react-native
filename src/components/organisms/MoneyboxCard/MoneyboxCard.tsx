import { StyleProp, View, ViewStyle } from 'react-native';

import {
	Button,
	Card,
	Flex,
	FloatNum,
	Heading,
	Icon,
	ProgressBar,
	Text,
} from 'components/atoms';
import { colors } from 'theme';

import { styles } from './MoneyboxCard.styles';

export interface Props {
	title: string;
	goal: number;
	saved: number;
	currency: string;
	testID: string;
	style?: StyleProp<ViewStyle>;
	onPressWithdrawalButton: () => void;
	onPressExtendButton: () => void;
	onPressTopUpButton: () => void;
}

export const MoneyboxCard = ({
	title,
	saved,
	goal,
	currency,
	style,
	testID,
	onPressWithdrawalButton,
	onPressExtendButton,
	onPressTopUpButton,
}: Props) => {
	const isFinished = saved / goal >= 1;

	const onPressSecondaryButton = isFinished
		? onPressExtendButton
		: onPressTopUpButton;

	return (
		<Card direction="column" style={[styles.container, style]} testID={testID}>
			<Flex direction="row">
				<View style={styles.iconContainer}>
					{isFinished ? (
						<View style={styles.iconBackground}>
							<Icon color={colors.white} name="check" size={14} />
						</View>
					) : (
						<Icon color={colors.bodyText} name="piggy-bank" size={27} />
					)}
				</View>
				<Text
					style={styles.goalTitle}
					testID="moneybox-goal-title-test-id"
					text={title}
				/>
				<Flex direction="row" style={styles.goal}>
					<FloatNum
						floatStyle={styles.floatStyle}
						intStyle={styles.intStyle}
						number={`${(goal / 100).toFixed(2)}`}
						testID="moneybox-goal-test-id"
					/>
					<Heading
						size="h5"
						style={styles.currency}
						testID="moneybox-goal-currency-test-id"
						text={currency}
					/>
				</Flex>
			</Flex>

			<ProgressBar
				maxValue={goal}
				style={styles.progressBar}
				testID="moneybox-progress-bar-test-id"
				value={saved}
			/>

			<Flex direction="row" style={styles.saved}>
				<FloatNum
					floatStyle={styles.savedFloat}
					intStyle={styles.savedInt}
					number={`${(saved / 100).toFixed(2)}`}
					style={styles.savedFloatNum}
					testID="moneybox-saved-test-id"
				/>

				<Text
					style={[styles.savedFloat, styles.savedCurrency]}
					testID="moneybox-saved-currency-test-id"
					text={currency}
				/>
			</Flex>

			<Flex direction="row">
				<Button
					style={isFinished ? styles.extendButton : styles.topUpButton}
					testID={`moneybox-${isFinished ? 'extend' : 'top-up'}-btn-test-id`}
					title={isFinished ? 'Extend' : 'Top - Up'}
					type="ternary"
					onPress={onPressSecondaryButton}
				/>
				<Button
					style={styles.withdrawalButton}
					testID="moneybox-withdrawal-btn-test-id"
					title="Withdrawal"
					type="ternary"
					onPress={onPressWithdrawalButton}
				/>
			</Flex>
		</Card>
	);
};
