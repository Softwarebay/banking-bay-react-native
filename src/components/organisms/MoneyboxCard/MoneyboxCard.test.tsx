import { fireEvent, render, screen } from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';

import { MoneyboxCard } from './MoneyboxCard';

import type { Props } from './MoneyboxCard';

describe('MoneyboxCard', () => {
	const onPressExtendButton = jest.fn<Props['onPressExtendButton'], []>();
	const onPressTopUpButton = jest.fn<Props['onPressTopUpButton'], []>();
	const onPressWithdrawalButton = jest.fn<
		Props['onPressWithdrawalButton'],
		[]
	>();
	const props: Omit<Props, 'goal' | 'saved'> = {
		testID: 'moneybox-card-test-id',
		currency: 'USD',
		title: 'Test',
		style: {
			marginBottom: 10,
		},
		onPressExtendButton,
		onPressTopUpButton,
		onPressWithdrawalButton,
	};
	const goalReachedProps: Props = {
		...props,
		goal: 500000,
		saved: 500000,
	};
	const goalNotReachedProps: Props = {
		...props,
		goal: 500000,
		saved: 100000,
	};

	it('should match snapshot - goal reached', () => {
		const tree = testRenderer
			.create(<MoneyboxCard {...goalReachedProps} />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot - goal not reached', () => {
		const tree = testRenderer
			.create(<MoneyboxCard {...goalNotReachedProps} />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should display moneybox title', () => {
		render(<MoneyboxCard {...goalReachedProps} />);

		const title = screen.getByTestId('goal-title-moneybox-card-test-id');

		expect(title).toHaveTextContent('Test');
	});

	it('should display moneybox saving goal value and saving currency', () => {
		render(<MoneyboxCard {...goalReachedProps} />);

		const goal = screen.getByTestId('goal-moneybox-card-test-id');
		const currency = screen.getByTestId('goal-currency-moneybox-card-test-id');

		expect(goal).toHaveTextContent('5000.00');
		expect(currency).toHaveTextContent('USD');
	});

	it("should display piggy-bank icon when saving goal isn't reached", () => {
		render(<MoneyboxCard {...goalNotReachedProps} />);

		const piggyIcon = screen.getByTestId('piggy-bank-icon-test-id');
		const checkIcon = screen.queryByTestId('check-icon-test-id');

		expect(piggyIcon).not.toBeNull();
		expect(checkIcon).toBeNull();
	});

	it('should display check icon when saving goal is reached', () => {
		render(<MoneyboxCard {...goalReachedProps} />);

		const checkIcon = screen.getByTestId('check-icon-test-id');
		const piggyIcon = screen.queryByTestId('piggy-bank-icon-test-id');

		expect(checkIcon).not.toBeNull();
		expect(piggyIcon).toBeNull();
	});

	it('should display full progress bar for when saving goal is reached', () => {
		render(<MoneyboxCard {...goalReachedProps} />);

		const progressLine = screen.getByTestId(
			'progress-line-progress-bar-moneybox-card-test-id'
		);

		expect(progressLine).toHaveStyle({
			width: '100%',
		});
	});

	it("should display not full progress bar for when saving goal isn't reached", () => {
		render(<MoneyboxCard {...goalNotReachedProps} />);

		const progressLine = screen.getByTestId(
			'progress-line-progress-bar-moneybox-card-test-id'
		);

		expect(progressLine).toHaveStyle({
			width: '20%',
		});
	});

	it('should display amount of saved money and money currency', () => {
		render(<MoneyboxCard {...goalNotReachedProps} />);

		const saved = screen.getByTestId('saved-moneybox-card-test-id');
		const currency = screen.getByTestId('saved-currency-moneybox-card-test-id');

		expect(saved).toHaveTextContent('1000.00');
		expect(currency).toHaveTextContent('USD');
	});

	it('should display two buttons', () => {
		render(<MoneyboxCard {...goalNotReachedProps} />);

		const buttons = screen.getAllByTestId(/-btn-moneybox-card-test-id/);

		expect(buttons).toHaveLength(2);
	});

	it("should render 'Withdrawal' and 'Extend' buttons when saving goal is reached", () => {
		render(<MoneyboxCard {...goalReachedProps} />);

		const withdrawalBtn = screen.getByTestId(
			'withdrawal-btn-moneybox-card-test-id'
		);
		const extendBtn = screen.getByTestId('extend-btn-moneybox-card-test-id');
		const topUpBtn = screen.queryByTestId('top-up-btn-moneybox-card-test-id');

		expect(withdrawalBtn).toHaveTextContent('Withdrawal');
		expect(extendBtn).toHaveTextContent('Extend');
		expect(topUpBtn).toBeNull();
	});

	it('should handle press events on both buttons when goal is reached', () => {
		render(<MoneyboxCard {...goalReachedProps} />);

		const withdrawalBtn = screen.getByTestId(
			'withdrawal-btn-moneybox-card-test-id'
		);
		const extendBtn = screen.getByTestId('extend-btn-moneybox-card-test-id');

		fireEvent.press(withdrawalBtn);
		expect(onPressWithdrawalButton).toHaveBeenCalledTimes(1);

		fireEvent.press(extendBtn);
		expect(onPressExtendButton).toHaveBeenCalledTimes(1);

		expect(onPressTopUpButton).not.toHaveBeenCalled();
	});

	it("should render 'Withdrawal' and 'Top - Up' buttons saving goal isn't reached", () => {
		render(<MoneyboxCard {...goalNotReachedProps} />);

		const withdrawalBtn = screen.getByTestId(
			'withdrawal-btn-moneybox-card-test-id'
		);
		const topUpBtn = screen.getByTestId('top-up-btn-moneybox-card-test-id');
		const extendBtn = screen.queryByTestId('extend-btn-moneybox-card-test-id');

		expect(withdrawalBtn).toHaveTextContent('Withdrawal');
		expect(topUpBtn).toHaveTextContent('Top - Up');
		expect(extendBtn).toBeNull();
	});

	it("should handle press events on both buttons when goal isn't reached", () => {
		render(<MoneyboxCard {...goalNotReachedProps} />);

		const withdrawalBtn = screen.getByTestId(
			'withdrawal-btn-moneybox-card-test-id'
		);
		const topUpBtn = screen.getByTestId('top-up-btn-moneybox-card-test-id');

		fireEvent.press(withdrawalBtn);
		expect(onPressWithdrawalButton).toHaveBeenCalledTimes(1);

		fireEvent.press(topUpBtn);
		expect(onPressTopUpButton).toHaveBeenCalledTimes(1);

		expect(onPressExtendButton).not.toHaveBeenCalled();
	});

	it('should apply additional styles', () => {
		render(<MoneyboxCard {...goalNotReachedProps} />);

		const card = screen.getByTestId('moneybox-card-test-id');

		expect(card).toHaveStyle({
			marginBottom: 10,
		});
	});
});
