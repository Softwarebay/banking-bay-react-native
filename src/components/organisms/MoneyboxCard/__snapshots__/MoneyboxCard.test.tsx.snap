// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoneyboxCard should match snapshot - goal not reached 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
      },
      Object {
        "flexDirection": "column",
      },
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "shadowColor": "#161966",
          "shadowOffset": Object {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.08,
          "shadowRadius": 16,
        },
        Array [
          Object {
            "alignItems": "stretch",
            "padding": 20,
          },
          Object {
            "marginBottom": 10,
          },
        ],
      ],
    ]
  }
  testID="moneybox-card-test-id"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
        },
        Object {
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignSelf": "flex-start",
          "marginRight": 10,
        }
      }
    >
      <SvgMock
        color="#4C4C60"
        height={27}
        testID="piggy-bank-icon-test-id"
        width={27}
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#4C4C60",
            "fontFamily": "MulishRegular",
            "fontSize": 17,
            "lineHeight": 27.2,
          },
          Object {
            "fontSize": 15,
          },
        ]
      }
      testID="moneybox-goal-title-test-id"
    >
      Test
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
          },
          Object {
            "flexDirection": "row",
          },
          Object {
            "marginLeft": "auto",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
            },
            Object {
              "flexDirection": "row",
            },
            Array [
              Object {
                "alignItems": "baseline",
              },
              undefined,
            ],
          ]
        }
        testID="moneybox-goal-test-id"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#1B1D4D",
              },
              Object {
                "alignSelf": "flex-start",
                "fontFamily": "MulishRegular",
                "fontSize": 22,
                "lineHeight": 22.1,
              },
            ]
          }
        >
          5000.
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#1B1D4D",
              },
              Object {
                "fontFamily": "MulishRegular",
                "fontSize": 17,
                "lineHeight": 22.1,
              },
            ]
          }
        >
          00
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#1B1D4D",
            },
            Object {
              "fontFamily": "MulishSemiBold",
              "fontSize": 17,
              "lineHeight": 22.1,
            },
            Object {
              "fontFamily": "MulishBold",
              "marginLeft": 5,
            },
          ]
        }
        testID="moneybox-goal-currency-test-id"
      >
        USD
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
    testID="moneybox-progress-bar-test-id"
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 10,
            "height": 5,
          },
          Object {
            "left": 0,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "backgroundColor": "#3EB290",
            "width": "20%",
          },
        ]
      }
      testID="progress-line-moneybox-progress-bar-test-id"
    />
    <View
      style={
        Array [
          Object {
            "borderRadius": 10,
            "height": 5,
          },
          Object {
            "backgroundColor": "#4C4C60",
            "opacity": 0.2,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
        },
        Object {
          "flexDirection": "row",
        },
        Object {
          "justifyContent": "center",
          "marginBottom": 10,
          "marginTop": 5,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
          },
          Object {
            "flexDirection": "row",
          },
          Array [
            Object {
              "alignItems": "baseline",
            },
            Object {
              "marginRight": 5,
            },
          ],
        ]
      }
      testID="moneybox-saved-test-id"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#1B1D4D",
            },
            Object {
              "color": "#4C4C60",
              "fontSize": 15,
            },
          ]
        }
      >
        1000.
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#1B1D4D",
            },
            Object {
              "color": "#4C4C60",
              "fontSize": 12,
            },
          ]
        }
      >
        00
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#4C4C60",
            "fontFamily": "MulishRegular",
            "fontSize": 17,
            "lineHeight": 27.2,
          },
          Array [
            Object {
              "color": "#4C4C60",
              "fontSize": 12,
            },
            Object {
              "top": 1,
            },
          ],
        ]
      }
      testID="moneybox-saved-currency-test-id"
    >
      USD
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
        },
        Object {
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          false,
          Object {
            "alignItems": "center",
            "borderRadius": 10,
            "height": 55,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "#3EB290",
            "borderWidth": 0,
            "height": 40,
          },
          Object {
            "backgroundColor": "#55ACEE",
            "flex": 1,
            "marginRight": 5,
          },
        ]
      }
      testID="moneybox-top-up-btn-test-id"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "MulishSemiBold",
              "fontSize": 17,
            },
            Object {
              "fontSize": 15,
            },
          ]
        }
      >
        Top - Up
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          false,
          Object {
            "alignItems": "center",
            "borderRadius": 10,
            "height": 55,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "#3EB290",
            "borderWidth": 0,
            "height": 40,
          },
          Object {
            "flex": 1,
            "marginLeft": 5,
          },
        ]
      }
      testID="moneybox-withdrawal-btn-test-id"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "MulishSemiBold",
              "fontSize": 17,
            },
            Object {
              "fontSize": 15,
            },
          ]
        }
      >
        Withdrawal
      </Text>
    </View>
  </View>
</View>
`;

exports[`MoneyboxCard should match snapshot - goal reached 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
      },
      Object {
        "flexDirection": "column",
      },
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "shadowColor": "#161966",
          "shadowOffset": Object {
            "height": 6,
            "width": 0,
          },
          "shadowOpacity": 0.08,
          "shadowRadius": 16,
        },
        Array [
          Object {
            "alignItems": "stretch",
            "padding": 20,
          },
          Object {
            "marginBottom": 10,
          },
        ],
      ],
    ]
  }
  testID="moneybox-card-test-id"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
        },
        Object {
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignSelf": "flex-start",
          "marginRight": 10,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#3EB290",
            "borderRadius": 50,
            "height": 27,
            "justifyContent": "center",
            "width": 27,
          }
        }
      >
        <SvgMock
          color="#FFFFFF"
          height={14}
          testID="check-icon-test-id"
          width={14}
        />
      </View>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#4C4C60",
            "fontFamily": "MulishRegular",
            "fontSize": 17,
            "lineHeight": 27.2,
          },
          Object {
            "fontSize": 15,
          },
        ]
      }
      testID="moneybox-goal-title-test-id"
    >
      Test
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
          },
          Object {
            "flexDirection": "row",
          },
          Object {
            "marginLeft": "auto",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
            },
            Object {
              "flexDirection": "row",
            },
            Array [
              Object {
                "alignItems": "baseline",
              },
              undefined,
            ],
          ]
        }
        testID="moneybox-goal-test-id"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#1B1D4D",
              },
              Object {
                "alignSelf": "flex-start",
                "fontFamily": "MulishRegular",
                "fontSize": 22,
                "lineHeight": 22.1,
              },
            ]
          }
        >
          5000.
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#1B1D4D",
              },
              Object {
                "fontFamily": "MulishRegular",
                "fontSize": 17,
                "lineHeight": 22.1,
              },
            ]
          }
        >
          00
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#1B1D4D",
            },
            Object {
              "fontFamily": "MulishSemiBold",
              "fontSize": 17,
              "lineHeight": 22.1,
            },
            Object {
              "fontFamily": "MulishBold",
              "marginLeft": 5,
            },
          ]
        }
        testID="moneybox-goal-currency-test-id"
      >
        USD
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
    testID="moneybox-progress-bar-test-id"
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 10,
            "height": 5,
          },
          Object {
            "left": 0,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "backgroundColor": "#3EB290",
            "width": "100%",
          },
        ]
      }
      testID="progress-line-moneybox-progress-bar-test-id"
    />
    <View
      style={
        Array [
          Object {
            "borderRadius": 10,
            "height": 5,
          },
          Object {
            "backgroundColor": "#4C4C60",
            "opacity": 0.2,
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
        },
        Object {
          "flexDirection": "row",
        },
        Object {
          "justifyContent": "center",
          "marginBottom": 10,
          "marginTop": 5,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
          },
          Object {
            "flexDirection": "row",
          },
          Array [
            Object {
              "alignItems": "baseline",
            },
            Object {
              "marginRight": 5,
            },
          ],
        ]
      }
      testID="moneybox-saved-test-id"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#1B1D4D",
            },
            Object {
              "color": "#4C4C60",
              "fontSize": 15,
            },
          ]
        }
      >
        5000.
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#1B1D4D",
            },
            Object {
              "color": "#4C4C60",
              "fontSize": 12,
            },
          ]
        }
      >
        00
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#4C4C60",
            "fontFamily": "MulishRegular",
            "fontSize": 17,
            "lineHeight": 27.2,
          },
          Array [
            Object {
              "color": "#4C4C60",
              "fontSize": 12,
            },
            Object {
              "top": 1,
            },
          ],
        ]
      }
      testID="moneybox-saved-currency-test-id"
    >
      USD
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
        },
        Object {
          "flexDirection": "row",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          false,
          Object {
            "alignItems": "center",
            "borderRadius": 10,
            "height": 55,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "#3EB290",
            "borderWidth": 0,
            "height": 40,
          },
          Object {
            "backgroundColor": "#FF8A71",
            "flex": 1,
            "marginRight": 5,
          },
        ]
      }
      testID="moneybox-extend-btn-test-id"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "MulishSemiBold",
              "fontSize": 17,
            },
            Object {
              "fontSize": 15,
            },
          ]
        }
      >
        Extend
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          false,
          Object {
            "alignItems": "center",
            "borderRadius": 10,
            "height": 55,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "#3EB290",
            "borderWidth": 0,
            "height": 40,
          },
          Object {
            "flex": 1,
            "marginLeft": 5,
          },
        ]
      }
      testID="moneybox-withdrawal-btn-test-id"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "MulishSemiBold",
              "fontSize": 17,
            },
            Object {
              "fontSize": 15,
            },
          ]
        }
      >
        Withdrawal
      </Text>
    </View>
  </View>
</View>
`;
